cpp/src/DataStorm/TopicFactoryI.h

```diff
@@ -63,13 +63,18 @@ namespace DataStormI
         void shutdown() const;
 
     private:
-        mutable std::mutex _mutex;
         std::weak_ptr<Instance> _instance;
-        std::shared_ptr<TraceLevels> _traceLevels;
-        std::map<std::string, std::vector<std::shared_ptr<TopicI>>> _readers;
-        std::map<std::string, std::vector<std::shared_ptr<TopicI>>> _writers;
+        mutable std::mutex _mutex;
         std::int64_t _nextReaderId;
         std::int64_t _nextWriterId;
+
+        // A map of topic readers indexed by the topic name.
+        // Each key is a topic name, and the corresponding value is a vector of readers for that topic.
```

Comment:
It's odd that `shared_ptr<TopicI>` represents a reader (resp. writer) for a Topic.